src/jinja2/parser.py

```diff
@@ -57,9 +57,12 @@ def __init__(
         name: t.Optional[str] = None,
         filename: t.Optional[str] = None,
         state: t.Optional[str] = None,
+        preserve_comments: bool = False,
     ) -> None:
         self.environment = environment
-        self.stream = environment._tokenize(source, name, filename, state)
+        self.stream = environment._tokenize(
+            source, name, filename, state, preserve_comments
+        )
```

Comment:
note: We only pass the `preserve_comments` value to the lexer, and do not store it as an instance attribute. Previously, I had `subparse` check `self.preserve_comments`, but I think  it's more robust to simply rely on what the lexer yields.